include/actions/AddFoamBCAction.h

```diff
@@ -2,6 +2,15 @@
 
 #include "InputParameters.h"
 #include "MooseObjectAction.h"
+#include "FoamProblem.h"
+
+template <typename T>
+inline void
+copyParamFromParam(InputParameters & dst, const InputParameters & src, const std::string & name_in)
+{
+  if (src.isParamValid(name_in))
+    dst.set<T>(name_in) = src.get<T>(name_in);
+}
```

Comment:
I would suggest putting this in a `namespace hippo::internal` to indicate that this is a private utility rather than part of Hippo's interface.

I would usually use `namespace detail`, like `boost`, but looks like MOOSE uses `internal`.